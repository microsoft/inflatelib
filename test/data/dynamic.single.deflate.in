
# Single block that outputs "THE QUICK BROWN FOX JUMPS OVER THE LAZY DOG" 100 times
# NOTE: This test is only applicable to Deflate
>1
1       # bfinal = true
10      # Compressed with dynamic codes

# The string we are repeating contains 43 characters. We use the following lengths/distance to write all of the text:
#
#   *   (4, 31) to repeat "THE " in the original string
#   *   (43, 43) to repeat the string once (2 total)
#   *   (86, 86) to repeat the string twice (4 total)
#   *   (129, 43) to repeat the string 3 times (7 total)
#   *   (172, 86) to repeat the string 4 times (11 total)
#   *   (215, 129) to repeat the string 5 times (16 total)
#   *   (258, 172) to repeat the string 6 times (22 total)
#
# The following was passed to 'block-encode' to generate the data below:
#
# "THE QUICK BROWN FOX JUMPS OVER "
# (4, 31)
# "LAZY DOG"
# (43, 43)
# (86, 86)
# (129, 43)
# (172, 86)
# (215, 129)
# (258, 172)
# 256

11100   # HLIT = 285 (28 + 257)
01110   # HDIST = 15 (14 + 1)
1100    # HCLEN = 16 (12 + 4)

# Code Length Alphabet Code Lengths:
011 101 011 011 000 000 000 010 000 011 000 100 000 101 000 011

# Literal/Length & Distance Alphabet Code Lengths:
>>1
110 >1 0010101 >>1
11110
110 >1 0010101 >>1
100
101 >1 00 >>1
1110
100
101 >1 00 >>1
101 >1 01 >>1
00
1110
00
00
100
00
00
100
00
101 >1 01 >>1
110 >1 1111111 >>1
110 >1 0010000 >>1
00
010
00
110 >1 0000100 >>1
00
11111 >1 000 >>1
00
010
00
010
00
100
100
11111 >1 110 >>1
011
011
010
011
010
011

# Literal/Length Tree:
#   Symbol      Bit Count   Code
#   ' '         3           000
#   'A'         5           01000
#   'B'         5           01001
#   'C'         5           01010
#   'D'         5           01011
#   'E'         4           0010
#   'F'         5           01100
#   'G'         5           01101
#   'H'         5           01110
#   'I'         5           01111
#   'J'         5           10000
#   'K'         5           10001
#   'L'         5           10010
#   'M'         5           10011
#   'N'         6           110000
#   'O'         4           0011
#   'P'         6           110001
#   'Q'         6           110010
#   'R'         5           10100
#   'S'         6           110011
#   'T'         6           110100
#   'U'         5           10101
#   'V'         6           110101
#   'W'         6           110110
#   'X'         6           110111
#   'Y'         6           111000
#   'Z'         6           111001
#   END         6           111010
#   258         6           111011
#   274         6           111100
#   278         6           111101
#   280         6           111110
#   282         6           111111
#   283         5           10110
#   284         5           10111
#
# Distance Tree:
#   Symbol      Bit Count   Code
#   9           2           00
#   10          2           01
#   12          2           10
#   14          2           11

# Encoded Data:
>>1
110100 01110 0010 000               # "THE "
110010 10101 01111 01010 10001 000  # "QUICK "
01001 10100 0011 110110 110000 000  # "BROWN "
01100 0011 110111 000               # "FOX "
10000 10101 10011 110001 110011 000 # "JUMPS "
0011 110101 0010 10100 000          # "OVER "
111011 00 >1 110 >>1                # Repeat "THE "
10010 01000 111001 111000 000       # "LAZY "
01011 0011 01101                    # "DOG "
111100 >1 000 >>1 01 >1 1010 >>1    # Repeat 1 time
111101 >1 0011 >>1 10 >1 10101 >>1  # Repeat 2 times
111110 >1 1110 >>1 01 >1 1010 >>1   # Repeat 3 times
111111 >1 01001 >>1 10 >1 10101 >>1 # Repeat 4 times
10110 >1 10100 >>1 11 >1 000000 >>1 # Repeat 5 times
10111 >1 11111 >>1 11 >1 101011 >>1 # Repeat 6 times
111010                              # END
