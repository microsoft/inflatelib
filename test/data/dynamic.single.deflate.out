
repeat (22) {
    "THE QUICK BROWN FOX JUMPS OVER THE LAZY DOG"
}
